tests/check/check_userset.go

```diff
@@ -0,0 +1,106 @@
+			{Object: "directs-employee:userset_1", Relation: "direct", User: "employee:userset_valid"},
+
+			{Object: "usersets-user:userset_1", Relation: "userset", User: "directs-user:userset_1#direct"},
+			{Object: "usersets-user:userset_1", Relation: "userset", User: "directs-employee:userset_1#direct"},
+		},
+		CheckAssertions: []*checktest.Assertion{
+			{
+				Name:        "user_valid",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_1", Relation: "userset", User: "user:userset_valid"},
+				Expectation: true,
+			},
+			{
+				Name:        "employee_valid",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_1", Relation: "userset", User: "employee:userset_valid"},
+				Expectation: true,
+			},
+			{
+				Name:        "user_invalid",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_1", Relation: "userset", User: "user:userset_invalid"},
+				Expectation: false,
+			},
+			{
+				Name:        "employee_invalid",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_1", Relation: "userset", User: "employee:userset_invalid"},
+				Expectation: false,
+			},
+			{
+				Name:        "invalid_object",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_2", Relation: "userset", User: "user:userset_invalid"},
+				Expectation: false,
+			},
+		},
+	},
+	{
+		Name: "usersets_userset_recursive",
+		Tuples: []*openfgav1.TupleKey{
+			{Object: "usersets-user:userset_recursive_1", Relation: "userset_recursive", User: "user:userset_recursive_user_1"},
+			{Object: "usersets-user:userset_recursive_1", Relation: "userset_recursive", User: "usersets-user:userset_recursive_2#userset_recursive"},
+		},
+		CheckAssertions: []*checktest.Assertion{
+			{
+				Name:        "valid_recursive",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_recursive_1", Relation: "userset_recursive", User: "usersets-user:userset_recursive_2#userset_recursive"},
+				Expectation: true,
+			},
+			{
+				Name:        "valid_user",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_recursive_1", Relation: "userset_recursive", User: "user:userset_recursive_user_1"},
+				Expectation: true,
+			},
+			{
+				Name:        "invalid_recursive",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_1", Relation: "userset_recursive", User: "usersets-user:userset_3#userset_recursive"},
+				Expectation: false,
+			},
+			{
+				Name:        "invalid_user",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_1", Relation: "userset_recursive", User: "user:userset_user_2"},
+				Expectation: false,
+			},
+
+			{
+				Name:        "invalid_object",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:userset_2", Relation: "userset_recursive", User: "user:userset_user_2"},
+				Expectation: false,
+			},
+		},
+	},
+	{
+		Name: "usersets_or_userset", // does not cover condition
```

Comment:
To make this test complete, we should also add the second child of the "or" (i.e., `userset_to_computed_cond`)

Reply:
Done as part of 9174e1c8025077ded136fa5547bf8573c303019d